lib/mumukit/templates/query_hook.rb

```diff
@@ -0,0 +1,12 @@
+module Mumukit::Templates
+  class QueryHook < FileHook
+    def post_process_file(_file, result, status)
+      error_patterns.each { |it| return it.transform(result, status) if it.matches? result } if status.failed?
```

Comment:
Could we extract this code to a mixin? I think there implementors of QueryHook that inherit from a different class than FileHook. 

I would like to be able to introduce this new behavior without requiring modifications to the inheritance chain

Reply:
Can't this: `it.transform(result, status) if it.matches? result` be a method of ErrorPattern?